### PR TITLE
fix(github): Prevent race condition IntegrityErrors

### PR DESCRIPTION
**Pull request by MeredithAnya**
Pretty sure the `IntegrityError`s we've been seeing here are from race conditions, so switched to using `get_or_create` to help prevent that.

### CODE REVIEW

File: src/sentry/integrations/github/webhook.py [hunk 1]
```diff
@@ -347,18 +347,14 @@ def _handle(self, integration, event, organization, repo, host=None):
...
-                    name=user["login"][:128],
-                )
-
+            author, created = CommitAuthor.objects.get_or_create(
+                organization_id=organization.id,
+                email=author_email,
+                defaults={"name": user["login"][:128]},
+            )
+            if created:
+                author.external_id = self.get_external_id(user["login"])
```
Comment (mgaeta) [inline src/sentry/integrations/github/webhook.py:356]: Why can't we pass `external_id` to `defaults`?

Comment (MeredithAnya) [reply src/sentry/integrations/github/webhook.py:356]: I thought about that, and wasn't sure if there was a possibility of overwriting the `external_id` but it kind of looks like it's always that format? (using the `github:<login_name>`)

Comment (mgaeta) [reply src/sentry/integrations/github/webhook.py:356]: The `defaults` are only used in the "create" branch so it should be safe to add it.

Comment (MeredithAnya) [reply src/sentry/integrations/github/webhook.py]: ah yeah true true, updated

**Review by mgaeta — APPROVED** [review]

**Pull request merged**